src/unstructured_client/_hooks/custom/split_pdf_hook.py

```diff
@@ -250,8 +273,17 @@ def before_request(
                 page_count % split_size,
             )
 
+        # Use a variable to adjust the httpx client timeout, or default to 30 minutes
+        # When we're able to reuse the SDK to make these calls, we can remove this var
+        # The SDK timeout will be controlled by parameter
+        client_timeout_minutes = 30
+        if timeout_var := os.getenv("UNSTRUCTURED_CLIENT_TIMEOUT_MINUTES"):
```

Comment:
does the client user have to know to set this environment variable?  if so should the default go the other way around maybe? just thinking about how it might be challenging waiting up to 30 minutes without realizing why it's hanging so long

Reply:
We've had some users get stuck here. For very large pdfs, a hi_res call can sit for 30 minutes, at which point they get a httpx.ReadTimeout, the doc gets retried, and the same thing happens again. This should be rare, but we wanted the timeout to account for the worst case. And we added the env var so we could ask users to bump the timeout manually if this still wasn't resolved.

There is a http timeout parameter for the SDK now, but due to some gaps in the hook context, we don't have access to that param from down here. Longer term, that should be the value that users can configure. The env var here is a bit of a bandaid, so I'm keeping it out of the readme, etc.